pkg/controller/injector/serializer_test.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package injector
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+)
+
+// Tests in this file test this package's serialize() function by "round
+// tripping." Values are serialized using the serialize() function, then are
+// deserialized using the inverse of the process used by serialize(). Comparison
+// of the original value to the output of the round trip is used to assert the
+// correcness of the serialize() function.
+
+func TestSerializeInt(t *testing.T) {
+	intVals := []int{-13, -8, -5, -3, -2, -1, 0, 1, 2, 3, 5, 8, 13}
+	for _, intVal := range intVals {
+		bytes, err := serialize(intVal)
+		if err != nil {
+			t.Fatalf("Unexpected error: %s", err)
+		}
+		var intValPrime int
+		err = json.Unmarshal(bytes, &intValPrime)
+		if err != nil {
+			t.Fatalf("Unexpected error: %s", err)
+		}
+		if intVal != intValPrime {
+			t.Fatalf("Round trip failed; expected %v; got %v", intVal, intValPrime)
+		}
+	}
+}
+
+func TestSerializeFloat(t *testing.T) {
+	floatVals := []float64{-3.1415926535, -2.7182818284, 2.7182818284, 3.1415926535}
+	for _, floatVal := range floatVals {
+		bytes, err := serialize(floatVal)
+		if err != nil {
+			t.Fatalf("Unexpected error: %s", err)
+		}
+		var floatValPrime float64
+		err = json.Unmarshal(bytes, &floatValPrime)
+		if err != nil {
+			t.Fatalf("Unexpected error: %s", err)
+		}
+		if floatVal != floatValPrime {
+			t.Fatalf("Round trip failed; expected %v; got %v", floatVal, floatValPrime)
+		}
+	}
+}
+
+func TestSerializeString(t *testing.T) {
+	strVals := []string{
+		"foo",
+		"bar",
+		"bat",
+		"bat",
```

Comment:
>Fuzzing seems like it would be overkill, no?

FWIW I agree that it's overkill _for now_. Until we have a fully-functioning product (controller that talks to API server, complete working demo), I'd rather focus time on making that work than beefing up unit tests. Maybe make an issue for adding a fuzz test here?